src/systems/thruster/Thruster.cc

```diff
@@ -475,6 +514,19 @@ void ThrusterPrivateData::OnCmdThrust(const gz::msgs::Double &_msg)
   this->propellerAngVel = this->ThrustToAngularVec(this->thrust);
 }
 
+/////////////////////////////////////////////////
+void ThrusterPrivateData::OnDeabandEnable(const gz::msgs::Boolean &_msg)
+{
+  if (_msg.data() != this->enableDeadband)
```

Comment:
I recommend to use a mutex to protect `this->enableDeadband`.

Reply:
is this really necessary? as the only places where there is write-access to `this->enableDeadband` is in this method and when loading the sdf (which is necessary before the callback exist). So I don't see any conflict. Or did I miss something?

Reply:
In practice, this is unlikely to be a problem.

However there are several issues at play here. First, callbacks from `gz-transport` are run on a separate thread. This means that when a message arrives and is processed there is no guarantee that it will run on the same CPU core. Thus if you don't have any synchronization primitive there is no guarantee that `this->enableDeadband` will be read correctly on the next round due to cache coherency issues (see: https://blog.the-pans.com/std-atomic-from-bottom-up/).